python/sglang/srt/managers/schedule_batch.py

```diff
@@ -930,6 +930,8 @@ def retract_decode(self):
 
         self.filter_batch(keep_indices=sorted_indices)
 
+        if self.spec_info:
```

Comment:
this is not needed. `self.filter_batch` already filters the requests. We can delete these two lines.